worlds/hk_rework/state_mixin.py

```diff
@@ -259,14 +270,14 @@ def GetTerms(cls):
         """"""
         return []
 
-    def ModifyState(self, state_blob, item_state, player):  # -> Generator["state_blob"]:
+    def ModifyState(self, state_blob: Counter, item_state: CollectionState, player: int) -> Counter:  # -> Generator["state_blob"]:
```

Comment:
since this yields multiple Counters i believe this typing is wrong
<https://docs.python.org/3/library/typing.html#annotating-generators-and-coroutines>
```suggestion
    def ModifyState(self, state_blob: Counter, item_state: CollectionState, player: int) -> Generator[Counter]:
```

also you used `Generator[Counter]` later for a different ModifyState line lol